clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -546,7 +574,7 @@ CIRGenModule::getOrCreateCIRGlobal(StringRef MangledName, mlir::Type Ty,
       assert(0 && "not implemented");
 
     // TODO(cir): check TargetAS matches Entry address space
-    if (Entry.getSymType() == Ty &&
+    if (isValidRedeclarationType(Entry.getSymType(), Ty) &&
```

Comment:
I believe this diverges from the original codegen. The `Entry.getSymType() == Ty` statement should be evaluated as `false`.